### PR TITLE
JCSP-416: DC, Cloud: Transition an issue

### PR DESCRIPTION
**Pull request by rsaliiatlassian**

### CODE REVIEW

**Comment by dagguh** [comment]
@rsaliiatlassian instead of `(Don't Review)`, you can create draft PRs.
I converted this one to draft.

File: custom-vu/src/main/kotlin/jces1209/vu/page/CloudIssuePage.kt [hunk 1]
```diff
@@ -60,6 +67,116 @@ class CloudIssuePage(
...
+            return false
+        }
+        return true
+    }
+
+    override fun cancelTimeSpentForm(): AbstractIssuePage {
+        driver
+            .wait(
+                timeout = Duration.ofSeconds(5),
+                condition = ExpectedConditions.presenceOfElementLocated(By.id("issue-workflow-transition-cancel"))
```
Comment (ysemenkov) [inline custom-vu/src/main/kotlin/jces1209/vu/page/CloudIssuePage.kt]: Please extract and reuse locator
val transitionCancelLocator = By.id("issue-workflow-transition-cancel")

File: custom-vu/src/main/kotlin/jces1209/vu/page/CloudIssuePage.kt [hunk 2]
```diff
@@ -60,6 +67,116 @@ class CloudIssuePage(
...
+            .wait(
+                timeout = Duration.ofSeconds(7),
+                condition = ExpectedConditions.presenceOfElementLocated(By.xpath("//*[@data-test-id = 'issue.views.issue-base.foundation.status.status-field-wrapper']//*[contains(@data-test-id,\"issue.fields.status.common.ui.status-lozenge\")]"))
+            )
+            .click()
+        return this
+    }
+
+    private fun waitForPage() {
+        val executor = driver as JavascriptExecutor
```
Comment (ysemenkov) [inline custom-vu/src/main/kotlin/jces1209/vu/page/CloudIssuePage.kt]: What about to reuse FalliblePage.waitForPageToLoad?

File: custom-vu/src/main/kotlin/jces1209/vu/page/CloudIssuePage.kt [hunk 3]
```diff
@@ -60,6 +67,116 @@ class CloudIssuePage(
         return CloudIssueLinking(driver)
     }
 
+    override fun isTimeSpentFormAppeared(): Boolean {
+        try {
+            driver
```
Comment (ysemenkov) [inline custom-vu/src/main/kotlin/jces1209/vu/page/CloudIssuePage.kt:71]: What about jut to use ExpectedConditions.or()? so we should not wait until timeout exception

File: custom-vu/src/main/kotlin/jces1209/vu/page/DcIssuePage.kt [hunk 4]
```diff
@@ -57,4 +63,78 @@ class DcIssuePage(
...
+            )
+        return this
+    }
+
+    override fun transition(): DcIssuePage {
+        waitForPage()
+        WebDriverWait(driver, 7)
+            .until {
+                try {
+                    driver
```
Comment (ysemenkov) [inline custom-vu/src/main/kotlin/jces1209/vu/page/DcIssuePage.kt:130]: Why is it necessary to use such workaround?
Is element not clickable after `ExpectedConditions.elementToBeClickable`?

**Review by ysemenkov — CHANGES_REQUESTED** [review]

File: custom-vu/src/main/kotlin/jces1209/vu/page/DcIssuePage.kt (hunk 4 above)
Comment (rsaliiatlassian) [reply custom-vu/src/main/kotlin/jces1209/vu/page/DcIssuePage.kt:130]: Marked as TODO

**Pull request closed without merging**